Source/Core/DolphinQt/Config/ControllerInterface/DualShockUDPClientWidget.cpp

```diff
@@ -24,17 +26,20 @@ void DualShockUDPClientWidget::CreateWidgets()
 {
   auto* main_layout = new QGridLayout;
 
-  m_server_enabled = new QCheckBox(tr("Enable"));
-  m_server_enabled->setChecked(Config::Get(ciface::DualShockUDPClient::Settings::SERVER_ENABLED));
+  m_servers_enabled = new QCheckBox(tr("Enable"));
+  m_servers_enabled->setChecked(Config::Get(ciface::DualShockUDPClient::Settings::SERVERS_ENABLED));
+  main_layout->addWidget(m_servers_enabled, 0, 0);
+
+  m_server_list = new QListWidget();
+  main_layout->addWidget(m_server_list, 1, 0);
 
-  m_server_address = new QLineEdit(
-      QString::fromStdString(Config::Get(ciface::DualShockUDPClient::Settings::SERVER_ADDRESS)));
-  m_server_address->setEnabled(m_server_enabled->isChecked());
+  m_add_server = new QPushButton(tr("Add..."));
+  m_add_server->setEnabled(m_servers_enabled->isChecked());
+  main_layout->addWidget(m_add_server, 2, 0);
```

Comment:
I haven't checked if this compiles, but adding the buttons to an QHBoxLayout and then adding that QHBoxLayout to the main layout instead of adding the buttons directly to the main layout should fix the weird placement of the "Remove" button and produce similar results to the existing Game Config and USB Whitelist panels. Something along those lines: 
```suggestion
  QHBoxLayout* hlayout = new QHBoxLayout;
  hlayout->addStretch();

  hlayout->addWidget(m_add_server);
  hlayout->addWidget(m_remove_server);
  
  main_layout->addLayout(hlayout);
```

Reply:
Updated, better?

Reply:
Yes, much better. LGTM